curve_apps/parts_connection/driver.py

```diff
@@ -0,0 +1,197 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of edge-detection package.
+#
+#  All rights reserved.
+#
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+from __future__ import annotations
+
+import logging
+import sys
+
+import numpy as np
+from geoapps_utils.numerical import find_curves
+from geoh5py.groups import ContainerGroup
+from geoh5py.objects import Curve
+from geoh5py.ui_json import InputFile
+from tqdm import tqdm
+
+from ..driver import BaseCurveDriver
+from .params import Parameters
+
+logger = logging.getLogger(__name__)
+
+
+class PartsConnectionDriver(BaseCurveDriver):
+    """
+    Driver for the edge detection application.
+
+    :param parameters: Application parameters.
+    """
+
+    _parameter_class = Parameters
+
+    def __init__(self, parameters: Parameters | InputFile):
+        super().__init__(parameters)
+
+    def run(self):
+        """
+        Driver for Grid2D entity for the automated detection of line features.
+        The application relies on the Canny and Hough transforms from the
+        Scikit-Image library.
+        """
+        with self.workspace.open(mode="r+") as workspace:
+            parent = None
+            if self.params.output.out_group is not None:
+                parent = ContainerGroup.create(
+                    workspace=workspace,
+                    name=self.params.output.ga_group_name,
+                )
+
+            logger.info("Begin connecting labels ...")
+
+            vertices, cells, labels = self.get_connections()
+
+            logger.info("Process completed.")
+
+            if cells is None:
+                logger.info("No connections found.")
+                return
+
+            name = "Parts Connection"
+            if self.params.output.export_as is not None:
+                name = self.params.output.export_as
+
+            edges = Curve.create(
+                workspace=workspace,
+                name=name,
+                vertices=vertices,
+                cells=cells,
+                parent=parent,
+            )
+
+            if edges is not None:
```

Comment:
Not a big deal: should we create a function for this guy, to keep run light

Reply:
Done